flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/ScalarFunctionsTest.scala

```diff
@@ -4061,4 +4061,76 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "f55=f57",
       "true")
   }
+
+  @Test
+  def testStringFunctionAndExpressionResultType(): Unit = {
+    // this test is to check if the `resultType` of the `GeneratedExpression`
+    // of these string functions match their definition in `FlinkSqlOperatorTable`
+    // see FLINK-22106
```

Comment:
You don't need add JIRA number, it can be found in commit message.

Reply:
Indeed. But adding it in the comment is more straightforward.

Reply:
I still don't think we should leave the JIRA number here. Leaving the JIRA number in the code should be because the JIRA has not been resolved.
Here you should put the information in JIRA clearly here instead of JIRA number.